library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt

```diff
@@ -47,15 +50,20 @@ class AnalyticsEvent {
     val attributes: Map<String, String>
 
     @RestrictTo(RestrictTo.Scope.TESTS)
-    constructor(delay: Int = 0) {
-        action = null
+    constructor(
+        action: String? = null,
+        delay: Int = 0,
+        systems: Set<AnalyticsSystem>? = null,
+        attributes: Map<String, String>? = null
+    ) {
+        this.action = action
         this.delay = delay
-        systems = emptySet()
+        this.systems = systems.orEmpty()
         trigger = Trigger.DEFAULT
-        attributes = emptyMap()
+        this.attributes = attributes.orEmpty()
     }
 
-    internal constructor(parser: XmlPullParser) {
+    internal constructor(parent: Base, parser: XmlPullParser) {
```

Comment:
Ah, so we need the ```parent``` to know which tool the event is coming from?

Reply:
yeah, the parent gives us access to the parsed object hierarchy. The Manifest at the root of the hierarchy has the tool and locale